benchexec/tools/swat.py

```diff
@@ -0,0 +1,43 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.result as result
+from benchexec.tools.template import BaseTool2
+
+
+class Tool(BaseTool2):
+    """
+    Tool info module for SWAT a tool ensemble for the dynamic symbolic execution of modern Java.
+    """
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("run-swat.sh")
+
+    def version(self, executable):
+        return '42'
```

Comment:
Hard-coding a version number in this module is not a good idea. If people run a newer version of your tool with BenchExec they will get a wrong version number in the output, which could be very confusing.

The best would be to add a parameter like `--version` to your tool and read and return the actual version number here. You can call the utility method `self._version_from_tool` to do this easily.

If this is not possible, please remove the method completely. (But note that if you want to participate in SV-COMP or Test-Comp, AFAIK the rules there require a proper version number.)

Reply:
It should be no problem to add a proper version flag.  
However, would it be possible to leave it as it is for the pre-run? I have tested the current submission a few times and have already uploaded it to Zenodo. I am afraid that any last minute changes would break it.

But I can assure you that we will fix this for the final submission.

Reply:
I've just read the other comments. I'll just remove / comment the function for the Pre-Run, if that is okay with you?

Reply:
A hard-coded version number is not acceptable for BenchExec. A missing one is suboptimal, but is acceptable if the tool does not have a version number.

Everything else is not the concern of BenchExec, and would need to be discussed elsewhere if necessary.